benchmark/scripts/models/timehistory.py

```diff
@@ -8,11 +8,69 @@
 
 
 class TimeHistory(keras.callbacks.Callback):
-    def on_train_begin(self, logs={}):
+    """Callback that extract execution time of every epoch, time-step,
+    and speed in terms of sample per sec
+    """
+
+    def __init__(self):
+        super(TimeHistory, self).__init__()
+        self.times = []
+
+    def on_train_begin(self, logs=None):
         self.times = []
 
-    def on_epoch_begin(self, batch, logs={}):
+    def on_epoch_begin(self, batch, logs=None):
         self.epoch_time_start = time.time()
 
-    def on_epoch_end(self, batch, logs={}):
+    def on_epoch_end(self, batch, logs=None):
         self.times.append(time.time() - self.epoch_time_start)
+
+    def get_num_samples(self):
+        if 'samples' in self.params:
+            return self.params['samples']
+        elif 'steps' in self.params:
+            return self.params['steps']
+        else:
+            raise ValueError('Incorrect metric parameter')
+
+    def reformat(self, var):
```

Comment:
please make this a private method & limit its scope or move it to utils such that LoggingMetrics class can take advantage of it while appending to pattern list